analysis/src/SharedTypes.ml

```diff
@@ -296,6 +294,36 @@ end
 
 type polyVariantConstructor = {name: string; args: Types.type_expr list}
 
+(** An type that can be used to drive completion *)
+type completionType =
```

Comment:
How about factoring out `env`? Seems to be present in every case.

Reply:
Yes, that's a good idea. I'll need to have a deep look at how env is passed around, where it's needed, etc. Ok with you if I do that in a separate PR?